app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsRepository.scala

```diff
@@ -34,9 +32,11 @@ import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
+//TODO: think about getting rid of trait
 @ImplementedBy(classOf[SubscriptionFieldsMongoRepository])
 trait SubscriptionFieldsRepository {
 
+  //TODO consider Future[Boolean]
```

Comment:
Sounds good, as we do it also for the `delete`.
Alternatively you could make both method returning `Option[SubscriptionFields]`

Reply:
`Future[Boolean]` has already been implemented in PR5.